storage/innobase/fil/fil0fil.cc

```diff
@@ -3922,18 +3922,6 @@ void Fil_shard::close_all_files() {
 
 /** Close all open files. */
 void Fil_system::close_all_files() {
-#if defined UNIV_DEBUG || defined UNIV_BUF_DEBUG
-  bool should_validate_space_reference_count = srv_fast_shutdown == 0;
-  DBUG_EXECUTE_IF("buf_disable_space_reference_count_check",
-                  should_validate_space_reference_count = false;);
-
-  if (should_validate_space_reference_count) {
-    auto buffer_pool_references = buf_LRU_count_space_references();
-    for (auto shard : m_shards) {
-      shard->validate_space_reference_count(buffer_pool_references);
-    }
-  }
```

Comment:
1. why do we need to remove this code? 
2. what is the value of srv_fast_shutdown for PXB (either prepare or backup)?
3. Can this value be changed?

Reply:
> 1) why do we need to remove this code?
there was crash during the backup stage because mutex initialisation.  IIUC there should not be anything in buffer pool during backup stage. so I removed this check

2) for backup it is 0 and for prepare it is 1.

Reply:
For 1) Since the ref count is for dropping tablespaces, it is not relevant to PXB.  But you could have disabled with ifndef XTRBACKUP instead of removing it

For 2) Do u know the reason for the difference in value? is it set in PXB code?

Reply:
fixed 1.
it is set in PXB code. for backup we use default value of 0. but incase of prepare we set explicit  [at](https://github.com/percona/percona-xtrabackup/blob/fea8a0ef0e29a010972e944e24536f638c43c9bb/storage/innobase/xtrabackup/src/xtrabackup.cc#L2346). I am not sure why